### PR TITLE
[NCL-4982] Implement Build Endpoint Test

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

File: demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java [hunk 1]
```diff
@@ -534,12 +534,15 @@ public void initiliazeBuildRecordDemoData() {
                     .dependency(importedArtifact1)
                     .dependency(importedArtifact2)
                     .user(demoUser)
+                    .repourLog("This is a wanabe alignment log.")
```
Comment (michalovjan) [inline demo-data/src/main/java/org/jboss/pnc/demo/data/DatabaseDataInitializer.java]: *wannabe

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java [hunk 2]
```diff
@@ -96,42 +165,119 @@ public void shouldSetDependentArtifacts() throws RemoteResourceException {
         Assertions.assertThat(updatedArtifactIds).contains(102);
     }
 
+    @Test
+    public void shouldGetSCMArchiveLink() throws ClientException, ReflectiveOperationException {
+        BuildClient client = new BuildClient(RestClientConfiguration.asAnonymous());
+
+        // Disable redirects so we can test the actual response
+        Field f =  ClientBase.class.getDeclaredField("client"); //NoSuchFieldException
```
Comment (michalovjan) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java]: Do the Exceptions in comments have some meaning in the context of this test?

Comment (thescouser89) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildEndpointTest.java]: Grammar police

**Comment by janinko** [comment]
retest this please

**Pull request merged**